### PR TITLE
feature ready for PR

### PR DESCRIPTION
**Pull request by mariano98montero**
Reservation storing 
added funcionality for storing the reservation in a class database.

Package Tree:
image
image

device: Pixel 3a:
image
image
image

device: Nexus S:
image
image
image

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystemjava/entity/Reservation.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.example.parkingsystemjava.entity;
+
+public class Reservation {
+    private String entryDate;
+    private String exitDate;
+    private String keyCode;
+
+    public Reservation() {
+        this.entryDate = null;
+        this.exitDate = null;
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/parkingsystemjava/entity/Reservation.java]: pls use a empty string instead a null

File: app/src/main/java/com/example/parkingsystemjava/mvp/contract/ParkingReservationContract.java [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
     interface ReservationActivityView {
-        void showEntryDatePicker();
+        void showDatePicker(ListenerDateTime listenerDateTime);
+        void showDateSelected(String date);
+        void showConfirmationMessage(int parkingLot);
     }
 
     interface ReservationActivityModel {
+        void addReservation(Reservation reservation, int parkingLot);
     }
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/parkingsystemjava/mvp/contract/ParkingReservationContract.java]: Pls rename the contracts alligned oof the father interface

File: app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java [hunk 3]
```diff
@@ -1,17 +1,49 @@
...
+    @Override
+    public void setEntryExitDate(Calendar date) {
+        String stringDate = getStringFromDate(date);
+        view.showDateSelected(stringDate);
+    }
+
+    private String getStringFromDate(Calendar calendar) {
+        Date date = calendar.getTime();
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("hh: mm a dd-MMM-yyyy");
+        String inActiveDate = null;
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java]: use empty string instead the null

File: app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java [hunk 4]
```diff
@@ -1,17 +1,49 @@
...
+        try {
+            inActiveDate = simpleDateFormat.format(date);
+            System.out.println(inActiveDate);
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java]: this its not neccessary, insteand off use Log library to print in the console a message

File: app/src/main/java/com/example/parkingsystemjava/entity/Reservation.java (hunk 1 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/entity/Reservation.java]: done

File: app/src/main/java/com/example/parkingsystemjava/mvp/contract/ParkingReservationContract.java (hunk 2 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/contract/ParkingReservationContract.java]: done

File: app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java (hunk 3 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java]: done

File: app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java (hunk 4 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java]: done

File: app/src/main/java/com/example/parkingsystemjava/mvp/model/ParkingModel.java [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
 import com.example.parkingsystemjava.mvp.contract.ParkingContract;
 
 public class ParkingModel implements ParkingContract.MainActivityModel {
     private String parkingLotsAvailable;
+    private ReservationDatabase database;
 
     @Override
     public void setParkingLots(String parkingLotsAvailable) {
-        this.parkingLotsAvailable = parkingLotsAvailable;
+        database = ReservationDatabase.getInstance();
```
Comment (lucianoeli) [inline app/src/main/java/com/example/parkingsystemjava/mvp/model/ParkingModel.java]: should not instantiate the database here.
this couldnt be tested. think in Dependency Inversion.
This should be passed as a parameter in the constructor (injected)

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingReservationView.java [hunk 6]
```diff
@@ -2,22 +2,47 @@
...
-        DateReservationDialogFragment entryReservationDialogFragment = DateReservationDialogFragment.newInstance();
+    public void showDatePicker(ListenerDateTime listenerDateTime) {
+        DateReservationDialogFragment entryReservationDialogFragment = DateReservationDialogFragment.newInstance(listenerDateTime);
         FragmentManager fragmentManager = getFragmentManager();
         entryReservationDialogFragment.show(fragmentManager, DIALOG_FRAGMENT_ENTRY_DATE_PICKER);
     }
+
+    @Override
+    public void showDateSelected(String date) {
+        if (binding.editTextReservationActivityEntry.getText().toString().equals("")) {
```
Comment (lucianoeli) [inline app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingReservationView.java]: dont use harcodes, use constants.
or better, you can try with .isEmpty()

Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingReservationView.java]: done

File: app/src/main/java/com/example/parkingsystemjava/mvp/model/ParkingModel.java (hunk 5 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/model/ParkingModel.java]: done

File: app/src/main/java/com/example/parkingsystemjava/entity/Reservation.java [hunk 7]
```diff
@@ -6,8 +6,8 @@
     private String keyCode;
 
     public Reservation() {
-        this.entryDate = null;
-        this.exitDate = null;
+        this.entryDate = "";
+        this.exitDate = "";
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/parkingsystemjava/entity/Reservation.java]: pls not use hardcodes

File: app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java [hunk 8]
```diff
@@ -37,10 +38,10 @@ public void setEntryExitDate(Calendar date) {
     private String getStringFromDate(Calendar calendar) {
         Date date = calendar.getTime();
         SimpleDateFormat simpleDateFormat = new SimpleDateFormat("hh: mm a dd-MMM-yyyy");
-        String inActiveDate = null;
+        String inActiveDate = "";
         try {
             inActiveDate = simpleDateFormat.format(date);
-            System.out.println(inActiveDate);
+            Log.i("recentDateSetted", inActiveDate);
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java]: remember, NOT use hardcodes

File: app/src/main/java/com/example/parkingsystemjava/entity/Reservation.java (hunk 7 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/entity/Reservation.java]: solved

File: app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java (hunk 8 above)
Comment (mariano98montero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java]: done

**Review by mauroLonghi — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Pull request merged**